fredericoo/saltplay
# feat: implement sentry

# 👀 Overview

This PR will add Sentry support for our API routes, back-end and front-end. Errors will be logged on our free Sentry account, and if in production, a new ticket will be created on Linear.

## 🚀 What we have done

- Added Sentry dependency
- Made every API endpoint exported `withSentry`
- Suppressed false positive warning that Sentry warns us about

## 🏁 How to test

- Ask me for the API token for Sentry
- Add this code snippet to anywhere on the app (I tried adding it to `Navbar.tsx`) to trigger an error:
  ```typescript
  <button onClick={()=>throw new Error('If this makes it to Linear we fuckin rock')}>Trigger error</button>
  ```
- Run the app locally and click on it.
- Check Linear in art 1 minute

## ✅ Checklist:

- I have created automated tests to replicate my manual testing automatically
